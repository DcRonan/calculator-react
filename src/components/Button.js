import PropTypes from 'prop-types';

const Button = key => <button type="button">{key}</button>;

Button.defaultProps = {
  key: '',
};

Button.propTypes = {
  key: PropTypes.string,
};

export default Button;
