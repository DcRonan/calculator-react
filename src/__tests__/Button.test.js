import renderer from 'react-test-renderer';
import { render, screen } from '@testing-library/react';
import Button from '../components/Button';

describe('checks if Button component renders correctly', () => {
  const handleClick = () => {
    '';
  };

  test('should render Button correctly', () => {
    const tree = renderer
      .create(<Button name="" handleClick={handleClick} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('should display correct value on the screen', () => {
    render(<Button name="+" handleClick={handleClick} />);
    const result = screen.getByText('+');

    expect(result).toBeInTheDocument();
  });
});
