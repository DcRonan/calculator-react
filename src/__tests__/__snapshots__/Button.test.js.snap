// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`checks if Button component renders correctly should render Button correctly 1`] = `
<button
  className="btn"
  onClick={[Function]}
  style={
    Object {
      "backgroundColor": "",
    }
  }
  type="button"
>
  
</button>
`;
